pages/stories/share-your-story.js

```diff
@@ -1,10 +1,123 @@
 import styled from 'styled-components'
+import { useState, useEffect } from 'react'
+import * as R from 'ramda'
+import { v4 as uuidv4 } from 'uuid'
+
+import client from '../../client'
+
+import useForm from '../../lib/useForm'
+import toCamelCase from '../../lib/toCamelCase'
+
 import Container from '../../components/Container'
+import { PurpleButton, CoralButton } from '../../components/Button'
+import Block from '../../components/Block'
+import { FormFieldWithLabel } from '../../components/Form'
+
+const GET_SHARE_STORY_FORM = `*[_type == "form" && slug.current == "share-your-story"][0]{
+  _type, 
+  title, 
+  body, 
+  subtitle, 
+  "inputsFromSanity": inputs[]->{ title, required, type },
+  confirmationText
+}`
 
 const ShareStoryStyled = styled(Container).attrs({
   className: '',
 })``
 
-const ShareStory = () => <ShareStoryStyled>ShareStory</ShareStoryStyled>
+const Title = styled.h2.attrs({
+  className: 'font-serif text-lg leading-base mb-2.5',
+})``
+
+const ShareStory = ({ body, inputsFromSanity, subtitle, confirmationText }) => {
+  const uploadStoryToSanity = (inputs, setInputs, initialState) => {
+    const story = {
+      _id: `drafts.${uuidv4()}`,
+      _type: 'story',
+      author: `${R.isEmpty(inputs.yourName) ? 'anonymous' : inputs.yourName}`,
+      email: `${R.isEmpty(inputs.yourEmail) ? 'anonymous' : inputs.yourEmail}`,
+      body: [
+        {
+          _type: 'block',
+          markDefs: [],
+          children: [{ _type: 'span', text: `${inputs.yourStory}`, marks: [] }],
+        },
+      ],
+    }
+
+    client
+      .create(story)
+      .then(() => {
+        // trigger confirmation modal here
+        setInputs(initialState)
+        toggleSubmissionModal(true)
+      })
+      .catch(e => console.log('error uploading story', e)) //eslint-disable-line
+  }
+
+  const [formCompleted, updateFormCompleted] = useState(false)
+  const [inputs, handleInputChange, handleSubmit] = useForm(
+    R.mergeAll(
+      R.map(input => R.objOf(toCamelCase(input.title), ''), inputsFromSanity),
+    ),
+    uploadStoryToSanity,
+  )
+
+  useEffect(
+    () =>
+      inputs.yourStory === ''
```

Comment:
Don't know if it's relevant in this case, but an empty string is falsey so you might not need the `=== ''` unless you need to distinguish from other falsey values